src/Components/quiz/play/Quiz.jsx

```diff
@@ -31,15 +45,26 @@ class Quiz extends Component {
     constructor(props) {
         super(props);
         this.state = {
-          userBalance: BALANCE,  // call getUserCash
+          userID: getUserID(),
+          userBalance: null,  // call getUserCash
+          userShares: [],
           formInput: []
         };
+
+        this.init();
+    }
+
+    // initializes values retrieved from database
+    init = async () => {
```

Comment:
no, that was causing issues because render is called before componentDidMount.

Reply:
Yeah, I'm aware that it's called before the render , but that should mean it just loads a few seconds later no other function should be relying on those states prior. The way you have it setup rn, init isn't awaited for so in the case there's a large number of symbols and there's a longer waiting time it could break getSymbol data. Just use a .then in getSymbolData.

Reply:
ended up just adjusting the thing that was to account for the extra load and put it back in componentDidMount